File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
import math
def solution(number, limit, power):
    
    def getDamage(number):
        result = 0
        for i in range(1, int(math.sqrt(number)+1)):
            if number % i == 0:
                if number // i == i:
                    result += 1
                else:
                    result += 2
        return result
    
    answer = 0
    for i in range(1, number+1):
        damage = getDamage(i)
        if damage > limit:
            answer += power
        else:
            answer += damage
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.*;

class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i=1; i<=number; i++){
            int damage = numberToDamage(i);
            answer += (damage > limit) ? power : damage;
        }
        return answer;
    }
    
    public static int numberToDamage(int number){
        int result = 0;
        for(int i=1; i<=(int)Math.sqrt(number); i++){
            if(number % i == 0){
                if(number / i == i){
                    result++;
                } else {
                    result += 2;
                }
            }
        }
        return result;
    }
}
    